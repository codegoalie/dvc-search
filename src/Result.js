import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Result = ({ roomType, resort, startDate, endDate, points }) => {
  return (
    <ResultContainer>
      <ResortDescription>
        <Icon>{abbreviationFor[resort] || "N/A"}</Icon>
        <RoomDescription>
          <RoomType>{roomType}</RoomType>
          <ResortName>{resort}</ResortName>
        </RoomDescription>
      </ResortDescription>
      <Dates>
        {startDate} &mdash; {endDate}
      </Dates>
      <Points>{points} points</Points>
    </ResultContainer>
  );
};

Result.propTypes = {
  roomType: PropTypes.string.isRequired,
  resort: PropTypes.string.isRequired,
  startDate: PropTypes.string.isRequired,
  endDate: PropTypes.string.isRequired,
  points: PropTypes.number.isRequired
};

export default Result;

const ResultContainer = styled.div`
  background: white;
  border-radius: 0.25rem;
  padding: 1rem;
  box-shadow: 4px 2px 14px rgba(255, 255, 255, 0.25);
  color: black;
  text-align: left;

  display: grid;
  grid-template-columns: 5fr 2fr 1fr;
  column-gap: 1rem;

  justify-content: space-between;
  font-size: 1.5rem;
  & + & {
    margin-top: 1.5rem;
  }

  @media (max-width: 768px) {
    display: flex;
    flex-direction: column;
  }
`;

const ResortDescription = styled.div`
  display: flex;
`;

const Icon = styled.div`
  display: inline-block;
  border-radius: 50%;
  background: black;
  color: white;
  width: 3rem;
  height: 3rem;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 1rem;
  font-weight: 700;
  flex-shrink: 0;
`;

const RoomDescription = styled.div`
  margin-left: 2rem;

  @media (max-width: 768px) {
    margin-left: 1rem;
  }
`;

const RoomType = styled.div`
  display: inline-block;
  font-weight: bold;
`;

const ResortName = styled.div`
  font-size: 1.25rem;
  font-weight: 300;
`;

const Dates = styled.div`
  @media (max-width: 768px) {
    margin: 1rem 0;
    text-align: right;
  }
`;

const Points = styled.div`
  @media (max-width: 768px) {
    text-align: right;
    font-weight: bold;
  }
`;

const abbreviationFor = {
  "Aulani, Disney Vacation Club Villas, Ko Olina, Hawaii": "AUL",
  "Bay Lake Tower at Disney's Contemporary Resort": "BLT",
  "Boulder Ridge Villas at Disney's Wilderness Lodge": "BRV",
  "Copper Creek Villas & Cabins at Disney's Wilderness Lodge": "CCR",
  "Disney's Animal Kingdom Villas": "AKV",
  "Disney's Beach Club Villas": "BCV",
  "Disney's BoardWalk Villas": "BWV",
  "Disney's Hilton Head Island Resort": "HHI",
  "Disney's Old Key West Resort": "OKW",
  "Disney's Polynesian Villas & Bungalows": "POLY",
  "Disney's Riviera Resort": "RR",
  "Disney's Saratoga Springs Resort & Spa": "SSR",
  "Disney's Vero Beach Resort": "VBR",
  "The Villas at Disney's Grand Californian Hotel & Spa": "GCAL",
  "The Villas at Disney's Grand Floridian Resort & Spa": "VGF"
};
