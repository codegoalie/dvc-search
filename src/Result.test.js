/* globals test expect */
import React from "react";
import { render } from "@testing-library/react";
import Result from "./Result";

const props = {
  resort: "Name",
  roomType: "One-Bedroom Villa",
  viewType: "Theme Park View",
  startDate: "2020-05-01",
  endDate: "2020-05-09",
  points: 120,
  handleAvailabilityClick: () => {},
  goalEndDate: new Date("2020-05-07"),
};

test("renders resort name", () => {
  const { getByText } = render(<Result {...props} />);
  const headerElement = getByText(props.resort);
  expect(headerElement).toBeInTheDocument();
});

test("renders room type", () => {
  const { getByText } = render(<Result {...props} />);
  const headerElement = getByText(new RegExp(props.roomType));
  expect(headerElement).toBeInTheDocument();
});

test("renders view type", () => {
  const { getByText } = render(<Result {...props} />);
  const headerElement = getByText(new RegExp(props.viewType));
  expect(headerElement).toBeInTheDocument();
});

test("renders points", () => {
  const { getByText } = render(<Result {...props} />);
  const headerElement = getByText(`${props.points} points`);
  expect(headerElement).toBeInTheDocument();
});

test("renders N/A abbreviation for unknown resort names", () => {
  let localProps = props;
  localProps.resort = "Unknown resort name";
  const { getByText } = render(<Result {...props} />);
  const headerElement = getByText("N/A");
  expect(headerElement).toBeInTheDocument();
});

test("renders VGF abbreviation for Grand Floridian", () => {
  let localProps = props;
  localProps.resort = "The Villas at Disney's Grand Floridian Resort & Spa";
  const { getByText } = render(<Result {...props} />);
  const headerElement = getByText("VGF");
  expect(headerElement).toBeInTheDocument();
});

test("renders VGF colored icon for Grand Floridian", () => {
  let localProps = props;
  localProps.resort = "The Villas at Disney's Grand Floridian Resort & Spa";
  const { container } = render(<Result {...props} />);
  expect(container.firstChild).toMatchSnapshot();
});

test("renders number of days extension", () => {
  const { getByText } = render(<Result {...props} />);
  const extensionElement = getByText(/\+2 days/);
  expect(extensionElement).toBeInTheDocument();
});
