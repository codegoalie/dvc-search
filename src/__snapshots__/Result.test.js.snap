// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders VGF colored icon for Grand Floridian 1`] = `
.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  width: 10rem;
  margin-top: 0.5rem;
  padding: 0.35rem 0.25rem 0.25rem;
  border-radius: 0.5rem;
  border: none;
  font-size: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: darkslateblue;
  color: floralwhite;
  box-shadow: 2px 1px 2px rgba(0,0,0,0.25);
  -webkit-transition: box-shadow 0.25s ease-in-out;
  transition: box-shadow 0.25s ease-in-out;
}

.c10:hover {
  box-shadow: 2px 1px 2px rgba(0,0,0,0.5);
  color: khaki;
}

.c1 {
  background: white;
  border-radius: 0.25rem;
  padding: 1rem;
  box-shadow: 4px 2px 14px rgba(255,255,255,0.25);
  color: black;
  text-align: left;
  display: grid;
  grid-template-columns: 5fr 2fr 1fr;
  -webkit-column-gap: 1rem;
  column-gap: 1rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-size: 1.5rem;
}

.c1 + .c0 {
  margin-top: 1.5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  border-radius: 50%;
  background: linear-gradient(160deg,hsl(335,76%,41%) 60%,hsl(335,76%,31%));
  color: white;
  border: 2px solid linear-gradient(160deg,hsl(335,76%,41%) 60%,hsl(335,76%,31%));
  width: 3rem;
  height: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1rem;
  font-weight: 700;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4 {
  margin-left: 2rem;
}

.c5 {
  display: inline-block;
  font-weight: bold;
}

.c6 {
  font-size: 1.25rem;
  font-weight: 300;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 .end-date {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9 {
  font-size: 1.75rem;
}

.c8 {
  display: inline-block;
  padding: 0.25rem 1rem;
  background-color: hsla(120,100%,78%,0.4);
  border-radius: 0.25rem;
  margin-top: 0.25rem;
}

@media (max-width:768px) {
  .c10 {
    height: 3rem;
    width: 100%;
  }
}

@media (max-width:768px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:768px) {
  .c4 {
    margin-left: 1rem;
  }
}

@media (max-width:768px) {
  .c7 {
    margin: 1rem 0;
    text-align: right;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
  }
}

@media (max-width:768px) {
  .c9 {
    text-align: right;
    font-weight: bold;
  }
}

<div
  class="sc-AxjAm c0 c1"
>
  <div
    class="c2"
  >
    <div
      class="c3"
      color="white"
    >
      VGF
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        One-Bedroom Villa
         
        — 
        Theme Park View
      </div>
      <div
        class="c6"
      >
        The Villas at Disney's Grand Floridian Resort & Spa
      </div>
    </div>
  </div>
  <div
    class="c7"
  >
    <div>
      Fri 5/1
       —
       
      <span
        class="end-date"
      >
        Sat 5/9
      </span>
    </div>
    <div
      class="c8"
    >
      +
      2
       day
      s
    </div>
  </div>
  <div
    class="c9"
  >
    120
     points
    <button
      class="c10"
    >
      Check availability →
    </button>
  </div>
</div>
`;
